test/src/test/java/aima/test/unit/search/uninformed/BidirectionalSearchTest.java

```diff
@@ -0,0 +1,50 @@
+package aima.test.unit.search.uninformed;
+
+import aima.core.environment.map2d.GoAction;
+import aima.core.environment.map2d.SimplifiedRoadMapOfPartOfRomania;
+import aima.core.environment.support.ProblemFactory;
+import aima.core.search.api.Problem;
+import aima.core.search.api.SearchForActionsFunction;
+import aima.core.search.basic.uninformed.BidirectionalSearch;
+import aima.core.search.basic.uninformed.BreadthFirstSearch;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Created by manthan on 1/3/17.
+ */
+@RunWith(Parameterized.class)
+public class BidirectionalSearchTest {
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> implementations() {
+        return Arrays.asList(new Object[][]{{"BidirectionalSearch"}});
+    }
+
+    @Parameterized.Parameter
+    public String searchFunctionName;
+
+    public <A, S> List<A> searchForActions(Problem<A, S> problem) {
+        SearchForActionsFunction<A, S> searchForActionsFunction = new BidirectionalSearch<A, S>();
+        return searchForActionsFunction.apply(problem);
+    }
+
+    @Test
+    public void testBidirectionalSearch() {
```

Comment:
I though to do so, but according to the previous practise followed as seen [here](https://github.com/aimacode/aima-java/blob/AIMA4e/test/src/test/java/aima/test/unit/search/uninformed/BreadthFirstSearchTest.java), I combined the two tests into one.